### PR TITLE
Product availability endpoint addition

### PR DESCRIPTION
**Pull request by dineshkumarj86**
## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Tests
[ ] Other
```

## Checklist
Have you done the following?
```
[X] Exported any new types in index.ts
[X] Merged master
[X] Run `npm run lint`
[X] Ensure the tests pass
```

## What's new?
Added Code For Product Availability API in BrightPearl

### CODE REVIEW

File: src/BrightpearlClient.ts [hunk 1]
```diff
@@ -249,6 +250,13 @@ class BrightpearlClient {
         return httpResponse.data.response;
     }
 
+    public async getProductAvailabilityByProductId(productId: number, includeOptional: string): Promise<ProductAvailability> {
+        const optionalParam = includeOptional ? `?includeOptional=${includeOptional}` : ""; 
+        const url = `${this.baseUrl}/${ENDPOINTS.warehouse.product_availability}/${productId}${optionalParam}`;
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: I think this will resolve as a double `//`

File: src/IBrightpearlClient.ts [hunk 2]
```diff
@@ -842,6 +843,28 @@ interface ProductChannel {
...
+    allocated: number;
+}
+
+interface WarehouseStock extends Stock {
+    byLocation?: AttributeMap<Stock>;
+}
+
+interface ProductAvailability {
+    total: Stock;
+    warehouse: AttributeMap<WarehouseStock>;
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: I think this is plural `warehouses` http://api-docs.brightpearl.com/warehouse/product-availability/get.html

File: src/IBrightpearlClient.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: Unnecessary white space added

File: src/IBrightpearlClient.ts [hunk 4]
```diff
@@ -842,6 +843,28 @@ interface ProductChannel {
...
 }
 
+interface Stock {
+    inStock: number;
+    onHand: number;
+    allocated: number;
+}
+
+interface WarehouseStock extends Stock {
+    byLocation?: AttributeMap<Stock>;
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: 👍

File: src/BrightpearlClient.ts (hunk 1 above)
Comment (dineshkumarj86) [reply src/BrightpearlClient.ts]: Have updated in endpoints.

File: src/IBrightpearlClient.ts (hunk 2 above)
Comment (dineshkumarj86) [reply src/IBrightpearlClient.ts]: Fixed

**Pull request merged**